packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -340,6 +355,67 @@ export function executeStepDependencyGraph<
       enqueueLeafSteps();
     }
 
+    /**
+     * Loads cached step data.
+     * @param stepCacheFilePath
+     * @param context
+     */
+    async function loadCacheForStep(
+      stepCacheFilePath: string,
+      context: TStepExecutionContext,
+    ) {
+      let status = StepResultStatus.FAILURE;
+      const entitiesPath = `${stepCacheFilePath}/entities`;
+      const relationshipsPath = `${stepCacheFilePath}/relationships`;
+
+      const { jobState, logger } = context;
+
+      let entitiesCount = 0;
+      await walkDirectory({
+        path: entitiesPath,
+        iteratee: async ({ filePath }) => {
+          const { entities } = await readGraphObjectFile<FlushedEntityData>({
+            filePath,
+          });
+
+          if (entities) {
+            entitiesCount += entities.length;
+            await jobState.addEntities(entities);
+            status = StepResultStatus.CACHED;
+          }
+        },
+      });
+
+      let relationshipCount = 0;
+      await walkDirectory({
+        path: relationshipsPath,
+        iteratee: async ({ filePath }) => {
+          const { relationships } =
+            await readGraphObjectFile<FlushedRelationshipData>({
+              filePath,
+            });
+
+          if (relationships) {
+            relationshipCount += relationships.length;
+            await jobState.addRelationships(relationships);
+            status = StepResultStatus.CACHED;
+          }
+        },
+      });
+
+      if (entitiesCount || relationshipCount) {
+        logger.info(
+          `Loaded ${entitiesCount} entities and  ${relationshipCount} relationship(s) from cache.`,
```

Comment:
Minor:

It is good practices to place all variables in the logger function's first arg instead of using string interpolation. String interpolated logs are slow to search.

Example of a fast log search term:

```
Loaded entities and relationship(s) from cache.
```

Example of a slow search term for which we will need to use the log search "contains" operator to find all instances of our log message. E.g. `~= "relationship(s) from cache."`

```
Loaded 10 entities and 10 relationship(s) from cache.
```

Suggestion:

```ts
logger.info({
  entitiesCount,
  relationshipCount
}, 'Loaded entities and relationship(s) from cache.');
```